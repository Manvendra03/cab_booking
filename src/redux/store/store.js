import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../slices/userSlice';
import fetchCabReducer from '../slices/cabSlice';
import reservationReducer from '../slices/reservationSlice';
import menuReducer from '../slices/menuSlice';

const store = configureStore({
  reducer: {
    fetchCab: fetchCabReducer,
    user: userReducer,
    reservation: reservationReducer,
    menu: menuReducer,
  },
});

export default store;
