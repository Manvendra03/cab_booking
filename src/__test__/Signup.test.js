import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store/store';
import SignUpComponent from '../pages/Signup';

describe('Delete Cab', () => {
  it('renders sign up component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <SignUpComponent />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Sign up form', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SignUpComponent />
        </BrowserRouter>
      </Provider>,
    );

    const header = screen.getByText(/Log in/i);
    expect(header).toBeInTheDocument();
  });
  it('renders sign up component and whows the already have an account test', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SignUpComponent />
        </BrowserRouter>
      </Provider>,
    );

    const subHead = screen.getByText(/Already have an account?/i);
    expect(subHead).toBeInTheDocument();
  });
});
