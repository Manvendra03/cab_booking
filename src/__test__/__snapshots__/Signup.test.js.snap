// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Cab renders sign up component 1`] = `
<div
  className="py-6 pt-20 space-y-8 bg-lime-50 h-screen overflow-auto"
>
  <h2
    className="flex justify-center text-3xl md:text-4xl text-lime-800 font-bold"
  >
    Sign Up
  </h2>
  <form
    className="w-full grid grid-cols-1 gap-1 mb-20 px-4 md:px-10 text-lime-800 lg:px-20 xl:px-40 md:gap-x-2"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="fullName"
      >
        Full Name:
        <input
          id="fullName"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="email"
      >
        Email:
        <input
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="email"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password:
        <input
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="confirmPassword"
      >
        Confirm Password:
        <input
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
        />
      </label>
    </div>
    <button
      className="bg-lime-700 w-full flex gap-2 items-center transition-all duration-300 text-white rounded-md justify-center text-2xl px-8 py-2"
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <p
    className="flex justify-center"
  >
    Already have an account?Â 
    <a
      className="text-blue-800 font-italic"
      href="/login"
      onClick={[Function]}
    >
      Log in
    </a>
  </p>
</div>
`;
